src/main/java/org/jenkinsci/plugins/GithubSecretStorage.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.jenkinsci.plugins;
+
+import hudson.model.User;
+import org.jfree.util.Log;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+public class GithubSecretStorage {
+
+    private GithubSecretStorage(){
+        // no accessible constructor
+    }
+
+    public static boolean contains(@Nonnull User user) {
+        return user.getProperty(GithubAccessTokenProperty.class) != null;
+    }
+
+    public static @CheckForNull String retrieve(@Nonnull User user) {
+        GithubAccessTokenProperty property = user.getProperty(GithubAccessTokenProperty.class);
+        if (property == null) {
+            Log.debug("Cache miss for username: " + user.getId());
+            return null;
+        } else {
+            Log.debug("Token retrieved using cache for username: " + user.getId());
+            return property.getAccessToken();
+        }
+    }
+
+    public static void put(@Nonnull User user, @Nonnull String accessToken) {
+        Log.debug("Populating the cache for username: " + user.getId());
```

Comment:
Better to use the `Log.log(Level.DEBUG, "Populating the cache for username: {0}", user.getId());` syntax here and in other places from the performance reasons. it is not that important for cold paths, but it still makes sense to avoid excessive string constructions

Reply:
Different log library from Jenkins-core regular one. No performance improvement to use explicit level.

Reply:
Heh, then this is fine (c)